File: home/dot_local/share/lazyvim/lua/lazyvim_capabilities/context.lua
local M = {}

function M.create()
	local paths = require("lazyvim_capabilities.paths")
	vim.env.HOME = paths.home
	vim.env.USERPROFILE = paths.home
	vim.env.XDG_CONFIG_HOME = paths.join(paths.home, ".config")
	local platform = require("lazyvim_capabilities.platforms")
	platform.configure_runtime()
	return { paths = paths, platform = platform, versions = require("lazyvim_capabilities.versions") }
end

return M

File: home/dot_local/share/lazyvim/capabilities/context.mjs
import { targetHome } from "../lib/paths.mjs";
import * as platform from "../lib/platforms/runtime.mjs";
import { versions } from "../lib/versions.mjs";

export function createCapabilityContext() {
  process.env.HOME = targetHome;
  process.env.USERPROFILE = targetHome;
  process.env.XDG_CONFIG_HOME = `${targetHome}/.config`;
  platform.configureRuntimeEnvironment();
  return Object.freeze({ platform, targetHome, versions });
}
